addons/osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsPipelineRegistry.java

```diff
@@ -69,25 +66,12 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
         }
     }
 
-    private Path clonePipelinesLibrary(String repoUrl, String repoRef) throws IOException {
-        log.log(Level.INFO, "Indexing contents from {0} using {1} ref",
-                new Object[]{repoUrl, repoRef});
-        Path targetPath = Files.createTempDirectory("pipeline-library");
-        log.info("Created " + targetPath);
-        ProcessBuilder builder = new ProcessBuilder()
-                .command("git", "clone", repoUrl, "--branch", repoRef,
-                         "--recursive", "--depth=1", "--quiet",
-                         targetPath.toString())
-                .inheritIO();
-        log.info("Executing: " + builder.command().stream().collect(Collectors.joining(" ")));
-        try {
-            int exitCode = builder.start().waitFor();
-            assert exitCode == 0 : "Process returned exit code: " + exitCode;
-        } catch (InterruptedException e) {
-            log.log(Level.WARNING, "Interrupted indexing process");
-            throw new IOException("Interrupted", e);
+    private Path resolvePipelinesPath() throws IOException {
```

Comment:
Why do you copy the zip to a temp folder and unzip it? Couldn't you access it directly using `ZipEntry`? Or even using a Zip file system? (https://docs.oracle.com/javase/7/docs/technotes/guides/io/fsp/zipfilesystemprovider.html)

Reply:
I get a `Provider "vfs" not installed` error if I do that :smile: